server/src/main/java/org/opensearch/index/merge/MergedSegmentTransferTracker.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.merge;
+
+import org.opensearch.common.annotation.ExperimentalApi;
+import org.opensearch.common.metrics.CounterMetric;
+import org.opensearch.core.index.shard.ShardId;
+import org.opensearch.index.IndexSettings;
+import org.opensearch.index.shard.AbstractIndexShardComponent;
+
+/**
+ * A component that tracks stats related to merged segment replication operations.
+ * This includes metrics for pre-copy(warm) invocations, failures, bytes transferred, and timing information.
+ *
+ * @opensearch.experimental
+ */
+@ExperimentalApi
+public class MergedSegmentTransferTracker extends AbstractIndexShardComponent {
```

Comment:
Why do we need the MergedSegmentTransferTracker class to extend AbstractIndexShardComponent?